### PR TITLE
Real array in all loops 

### LINKED ISSUE #30: Use real arrays
@elig0n found a problem in multiple areas: https://github.com/e-m-b-a/emba/pull/29

A quick check shows 56 areas we should take a look:
```
└─$ grep for.*\;\ do modules/* -R | grep -v "{" | wc -l
56
```
I will close the original pull request and we document it via this issue.

### PR DESCRIPTION
**Pull request by p4cx**
Fix #30 

All loops are now with real arrays. Also you can't now run modules with a space in the filename. It will be now checked (it would have broken, if you did that :D )

### CODE REVIEW

File: helpers/path.sh [hunk 1]
```diff
@@ -198,24 +198,24 @@ config_list() {
...
+            FILES="$FILES""$FIRMWARE_PATH""$(realpath "$LINE" 2>/dev/null )""\\n"
           else
-            FILES="$FILES""$LINE""\n"
+            FILES="$FILES""$LINE""\\n"
           fi
         done
       done
       if [[ -n "$FILES" ]]; then
-        echo -e "$FILES"
+        echo -e "$FILES" | sed -z '$ s/\n$//' | sort | uniq -u
```
Comment (m-1-k-3) [inline helpers/path.sh:218]: is sort -u not enough uniq?

**Comment by m-1-k-3** [comment]
Nice:
```
└─$ grep for.*\;\ do modules/* -R | grep -v "{" | wc -l
0

```

I will do some testing now ...

File: helpers/path.sh (hunk 1 above)
Comment (p4cx) [reply helpers/path.sh]: it is enough :D

**Comment by p4cx** [comment]
If binwalk extracts firmware and there are possibly two file-systems: emba can now work with it and find everything, even if it is hard-coded like "/var/test.cfg" <- it searches now and every time this pattern occurs, emba uses it (print on terminal or further analyzation). Works pretty good ;)

**Review by m-1-k-3 — APPROVED** [review]
Testcases running fine. Thank you for fixing all this stuff

**Pull request merged**